sketch.js

```diff
@@ -52,31 +52,44 @@ class Tree {
     return [this.root].concat(this.root.getAllChildren())
   }
 
+  getNearbyNodes(x, y, radius) {
+    let range = new Rectangle(x, y, radius, radius)
+    return qtree.query(range)
+  }
+
   findNearestNode(x, y) {
     // https://stackoverflow.com/questions/8864430/compare-javascript-array-of-objects-to-get-min-max
-    return this.getAllNodes().reduce((prev, curr) => prev.dist(x,y) < curr.dist(x,y) ? prev : curr)
+    //return this.getAllNodes().reduce((prev, curr) => prev.dist(x, y) < curr.dist(x, y) ? prev : curr)
+    let radius = 50
+    for (let i = 1; i <= 5; i++) {
```

Comment:
I'm not a fan of these hard coded values because we may want to change the size of the canvas (this also goes for my code, I was being lazy when I initially wrote it). I suggest passing in a max size to this function and say just doubling the current search radius until we find the nearest node or we hit the max (making sure we also check against the max radius itself).

Reply:
PS: I think you should start with a much smaller radius if you implement the doubling thing, something like 10 or so.